### ISSUE TITLE
Dynamically set name of headers on senders during runtime

### ISSUE THREAD

**Issue by MLenterman** [frankframework/frankframework#7730]
### Related Problem or Business Case

Simple use-case: api-key needs to be provided in a header when calling an endpoint. Different customers have different VPN-tunnel solutions or slightly different authentication requirements on the endpoint. The header name varies per customer and/or per solution. One solution needs the api-key to be in the `x-api-key` header, while another might need it to be in `x-opentunnel-api-key`.

### Describe the Feature

A mechanism to set the name of a header during runtime on senders. Currently it is not possible to set the name of a Param dynamically. For a Param to be used as header, the name also needs to be included in the `headersParams` attribute, which needs to be known during startup.

### Collaboration with External Companies

No, community resolution only

**Comment by nielsm5** [comment]
Can't you use `${a.property.name}` for the parameter name? Use `headerParams=${a.property.name}` and `<param name="${a.property.name}" value="something.else"/>`.

**Comment by Laurens-makel** [comment]
> Can't you use `${a.property.name}` for the parameter name? Use `headerParams=${a.property.name}` and `<param name="${a.property.name}" value="something.else"/>`.

That''s pretty much still a fixed value, I assume that @MLenterman would like to do something like this:

```
<Param sessionKey="headerValue">
    <Param name="headerName" sessionKey="headerName"/>
</Param>
```

**Comment by nielsm5** [comment]
Perhaps, but it's not possible to change the proxy settings runtime without closing, reconfiguring and re-opening the sender. So I'm hoping that's not the case :)

**Comment by alisihab** [comment]
It might not be ideal but could be a solution to the described simple use-case

![image](https://github.com/user-attachments/assets/595bbf63-8a76-4a7e-88d4-cfc471c73c68)

**Comment by MLenterman** [comment]
> Can't you use `${a.property.name}` for the parameter name? Use `headerParams=${a.property.name}` and `<param name="${a.property.name}" value="something.else"/>`.

Could have formulated the issue a littlebit more clearly. For my use-case, simply using properties seems to work fine!

**Comment by MLenterman** [comment]
Seemed like it was working, but it is actually not.

With a WebServiceSender I get the following error:
```
Thu Oct 31 10:46:01 CET 2024:ERROR:An error occurred [org.apache.xmlbeans.XmlException: error: Content is not allowed in trailing section.], see error log for details
```
Im not able to see anything out of the ordinary in Ladybug, logs, or SoapUI http/error logging. Only when I switch the WebserviceSender for a HttpSender I can actually get a hint of what the issue might be with the following error:
```
org.frankframework.core.SenderException: (IllegalArgumentException): (URISyntaxException) Illegal character in query at index 54: /mock/services/translate/generic/zds/BeantwoordVraag?${openforms2xxllnc.connections.beantwoord-vraag.auth-header-name}=secret1234
```

Im suspecting that the property in the `headerParams` attribute is not actually getting resolved (in time), and/or the property in param name is not getting resolved (in time). More likely the latter. This would explain why the param is being passed as a QUERY param instead of a HEADER param and why the unresolved property name is used as param name. 

In the case of the WebServiceSender it probably breaks on a different issue, as there are no query params with SOAP. Here i think the param's value is being appended to the soap message because of the POST method and the param not being recognized as a header param.

**Comment by MLenterman** [comment]
> It might not be ideal but could be a solution to the described simple use-case
> 
> ![image](https://private-user-images.githubusercontent.com/10929856/380754061-595bbf63-8a76-4a7e-88d4-cfc471c73c68.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzNjkxODAsIm5iZiI6MTczMDM2ODg4MCwicGF0aCI6Ii8xMDkyOTg1Ni8zODA3NTQwNjEtNTk1YmJmNjMtOGE3Ni00YTdlLTg4ZDQtY2ZjNDcxYzczYzY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMxVDEwMDEyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFmYTU4OGU3ZGQzNzcxZTcxNjkzMTIxNmZiNzMyZjk5YTcxMDgwZDkyZTFhYzMzMmUxZjA1NzU3M2ZjYmZhYzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ax7RznvFvtN61o2Pg8x-GxKnwBKgp2cNnFc8jDBW1eM)

Unfortunatly we don't know all the possible variations beforehand.

**Comment by nielsm5** [comment]
This has been solved in #7811
